gateway/reverse_proxy.go

```diff
@@ -1089,8 +1089,15 @@ func (p *ReverseProxy) handoverRequestToGraphQLExecutionEngine(roundTripper *Tyk
 
 		if isProxyOnly {
 			proxyOnlyCtx := reqCtx.(*GraphQLProxyOnlyContext)
-			header = proxyOnlyCtx.upstreamResponse.Header
-			httpStatus = proxyOnlyCtx.upstreamResponse.StatusCode
+			// There is a case in the proxy-only mode where the request can be handled
+			// by the library without calling the upstream.
+			// This is a valid query for proxy-only mode: query { __typename }
+			// In this case, upstreamResponse is nil.
+			// See TT-6419 for further info.
+			if proxyOnlyCtx.upstreamResponse != nil {
```

Comment:
Am I mistaken that there is a way to assign a variable but also check for equality in the same line?
Something like

if isProxyOnly && (proxyOnlyCtx := reqCtx.(*GraphQLProxyOnlyContext)) != nil {}

Basically an effort to reduce nesting? Maybe I am making things up. I tried to google, but I could have sworn I have seen it.

Reply:
@Aenimus It's possible but I think this makes it harder to read. One level of nesting is not that bad.